pkg/communication/communication_config.go

```diff
@@ -0,0 +1,140 @@
+package communication
+
+import (
+	"errors"
+	"fmt"
+
+	"encoding/json"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"net/url"
+)
+
+const (
+	defaultRemoteMediationServer       string = "/vmturbo/remoteMediation"
+	defaultRemoteMediationServerUser   string = "vmtRemoteMediation"
+	defaultRemoteMediationServerPwd    string = "vmtRemoteMediation"
+	defaultRemoteMediationLocalAddress string = "http://127.0.0.1"
+)
+
+type ServerMeta struct {
+	TurboServer string `json:"turboServer,omitempty"`
+}
+
+func (meta *ServerMeta) validateServerMeta() error {
+	if meta.TurboServer == "" {
+		return errors.New("Turbo Server URL is missing")
+	}
+	if _, err := url.ParseRequestURI(meta.TurboServer); err != nil {
+		return fmt.Errorf("Invalid turbo address url: %v", meta)
+	}
+	return nil
+}
+
+type WebSocketConfig struct {
+	LocalAddress      string `json:"localAddress,omitempty"`
+	WebSocketUsername string `json:"websocketUsername,omitempty"`
+	WebSocketPassword string `json:"websocketPassword,omitempty"`
+	ConnectionRetry   int16  `json:"connectionRetry,omitempty"`
+	WebSocketPath     string `json:"websocketPath,omitempty"`
+}
+
+func (wsc *WebSocketConfig) validateWebSocketConfig() error {
+	if wsc.LocalAddress == "" {
+		wsc.LocalAddress = defaultRemoteMediationLocalAddress
+	}
+	// Make sure the local address string provided is a valid URL
+	if _, err := url.ParseRequestURI(wsc.LocalAddress); err != nil {
+		return fmt.Errorf("Invalid local address url found in WebSocket config: %v", wsc)
+	}
+
+	if wsc.WebSocketPath == "" {
+		wsc.WebSocketPath = defaultRemoteMediationServer
+	}
+	if wsc.WebSocketUsername == "" {
+		wsc.WebSocketUsername = defaultRemoteMediationServerUser
+	}
+	if wsc.WebSocketPassword == "" {
+		wsc.WebSocketPassword = defaultRemoteMediationServerPwd
+	}
+	return nil
+}
+
+type RestAPIConfig struct {
```

Comment:
This config structure is for the RestAPI client, should be defined in that file or directory.